chromium_src/components/history/core/browser/history_service.cc

```diff
@@ -0,0 +1,18 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "src/components/history/core/browser/history_service.cc"
+
+namespace history {
+
+void HistoryService::GetKnownToSyncCount(
+    base::OnceCallback<void(HistoryCountResult)> callback) {
+  backend_task_runner_->PostTaskAndReplyWithResult(
+      FROM_HERE,
+      base::BindOnce(&HistoryBackend::GetKnownToSyncCount, history_backend_),
+      std::move(callback));
```

Comment:
shouldn't we need `base::BindPostTaskToCurrentDefault` for this callback? Otherwise, it will be called on backedn_task_runner_

Reply:
Yes, we need to have the callback running at the caller task. Thanks, fixed at https://github.com/brave/brave-core/pull/21623/commits/480908abc636ff0830901342b659218a1e7b3b9c .

Reply:
I don't quite follow why this is needed. PostTaskAndReply* family already has the desired behavior:
```
  // Posts |task| on the current TaskRunner.  On completion, |reply| is posted
  // to the sequence that called PostTaskAndReply().  On the success case,
 ```
 
 And in fact, documentation for `BindPostTaskToCurrentDefault` recommends using `PostTaskAndReply` instead

Reply:
Indeed. Thanks, reverted https://github.com/brave/brave-core/pull/21623/commits/a0db1b0bdf2fda2569fbd5a789719595a58d34dc